pkg/utils/workloadidentity/secret_test.go

```diff
@@ -0,0 +1,244 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package workloadidentity
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	securityv1alpha1 "github.com/gardener/gardener/pkg/apis/security/v1alpha1"
+)
+
+var _ = Describe("#Secret", func() {
+	var (
+		fakeClient      client.Client
+		secretName      = "foo"
+		secretNamespace = "namespace"
+		ctx             context.Context
+	)
+
+	BeforeEach(func() {
+		fakeClient = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+		ctx = context.TODO()
+	})
+
+	It("should not be able to create a secret because workload identity info is not set", func() {
+		secret, err := NewSecret(secretName, secretNamespace)
+		Expect(err).To(HaveOccurred())
+		Expect(secret).To(BeNil())
+		Expect(err.Error()).To(ContainSubstring("workload identity name is not set"))
+		Expect(err.Error()).To(ContainSubstring("workload identity namespace is not set"))
+		Expect(err.Error()).To(ContainSubstring("workload identity provider type is not set"))
```

Comment:
```suggestion
		Expect(err.Error()).To(And(
			ContainSubstring("workload identity name is not set"),
			ContainSubstring("workload identity namespace is not set"),
			ContainSubstring("workload identity provider type is not set"),
		))
```

Not necessary a change request, but you might like it.